architecture/overview.md

```diff
@@ -2,50 +2,54 @@
-4. The engine deposits a portion of the fees into the fee pool
-5. At some point, the funds in the fee pool are paid out to token holders
-6. The vAMM calculates the input and output prices when swapping and the Engine uses these to calculate position size
-7. The positional data that the vAMM returns is then stored in the Engine. Most of the data is user input, but the vAMM calculates the position size
-8. The vAMM calculates the funding payments, and to do so it receives pricing data from the on-chain oracle, which is implemented via the Pricefeed contract
+### Users
+
+As a decentralised protocol users are free to interact with the protocol as they wish. Users have two roles:
+
+* Traders
+* Liquidators
+
+Traders are users who take perpetual positions within the protocol. Liquidators are users who monitor the margin levels of users and perform liquidations appropriately.
 
 ### Margin Engine
 
-The Engine owns all of the contracts. This means that most of the methods that exist in the other contracts will error if the sender is not the Engine.
+The margin engine manages the positions of protocol users ans is permissoned to perform actions with the vAMMs on behalf of users. Traders use the margin engine contract to open and close positions. Additionally traders are able to deposit and remove margin from their margin account within the engine as appropriate.
 
-The Engine can open and close positions, which sends a message to the vAMM to fetch a price and alter the reserves. The positional data is then returned to the Engine and stored. Users can also add or remove margin from existing positions through the Margin Engine, and positions are liquidated via the Engine.
+The margin engine holds all collateral of the traders, however should a shortfall occur the margin engine is able to withdraw funds from the insurance fund in order to cover a shortfall.
 
-During these transactions, a fee is charged by the protocol. This fee is split and some of it is sent to the Fee Pool whilst some of the fee is sent to the Insurance Fund.
+During the settlement of funding payments the margin engine applies the funding payment to the margin account of users.
 
-The Engine also controls pausing of the vAMM via `SetOpen`, but shutdown is still controlled by the Insurance Fund
+As part of the fee collection the margin engine pays a portion of the trading fees into the insurance fund and fee pool.
 
 ### vAMM
 
 The vAMM acts as the mechanism for price discovery. The contract records the reserves of base and quote assets, and enables swapping input and output - returning a price to the engine. The vAMM also returns the size of the position that a user purchases, which is the amount of base asset.
 
-The vAMM is also in charge of settling the funding, and does so by querying the pricefeed contract to find the oracle price.
+The vAMM is also uses the pricefeed contract to find the oracle price during settlement of funding payments..

```

Comment:
'The vAMM also uses the pricefeed contract to find the oracle price during settlement of funding payments.'

reads better - removed 'is' and second full stop